addons/website/static/src/js/utils.js

```diff
@@ -239,6 +239,28 @@ function websiteDomain(self) {
     return ['|', ['website_id', '=', false], ['website_id', '=', websiteID]];
 }
 
+/**
+ * Checks if the 2 given URLs are the same, to prevent redirecting uselessly
+ * from one to another.
+ * It will consider naked URL and `www` URL as the same URL.
+ * It will consider `https` URL `http` URL as the same URL.
+ *
+ * @param {string} url1
+ * @param {string} url2
+ * @returns {Boolean}
+ */
+function isHTTPSorNakedDomainRedirection(url1, url2) {
+    try {
+        url1 = new URL(url1).host;
+        url2 = new URL(url2).host;
+    } catch {
+        // Incorrect URL, `false` URL..
+        return false;
+    }
+    return url1 === url2 ||
+           url1.replace(/^www\./, '') === url2.replace(/^www\./, '');
```

Comment:
Actually, I wonder if this is was right again since a URL in the form of `https://www.sub.hello.com` would still be striped of its `www.` but I guess it's ok since:
- This is a weird case.
- In that case, the redirection would not be done only if the user is on `sub.hello.com` trying to reach a website which uses `www.sub.hello.com`, even weirder case I guess.

Reply:
> a URL in the form of https://www.sub.hello.com would still be striped of its www.

That's the expected behavior, at least that's what I wanted, no matter how many subdomain there is, I always want to check without the leading www, so `https://www.sub.hello.com` should be "equal" to `https://sub.hello.com`

Reply:
Yeah, I guess it's not a real usecase anyway, I was more thinking outloud than saying it was a problem :wink: